### PR TITLE
Added custom manager and queryset support

### LINKED ISSUE #333: Unable to use custom managers or querysets
A common practice is to create a manager to do "soft" deletes on data when that data needs to be tracked.  Unfortunately instead of deriving from `models.Manager`, the manager I have to deal with looks like:

```
class SoftDeleteManager(manager.BaseManager.from_queryset(SoftDeleteQuerySet)):
    use_for_related_fields = True

    def get_queryset(self):
        return super(SoftDeleteManager, self).get_queryset().filter(active=True)
```
So it would be nice to be able to set the custom manager class that gets monkey patched.  Currently it has to derive from `models.Manager`, but it doesn't have too to be a valid manager.

Pull request is incoming

### PR DESCRIPTION
**Pull request by xuru**
Fixes #333 for custom managers
- Updated check for manager to support custom managers
- Added tests
- Added exception handling around importing custom managers

### CODE REVIEW

**Comment by Suor** [comment]
As far as I see from Django code, all managers descend from `BaseManager`, so cacheops might just switch to patch that instead of `Manager`. No extra code will be required.

**Comment by Suor** [comment]
Also test could be simplified - a noop custom QuerySet and anonymous Manager should do. Also shouldn't test internals in test body.

**Comment by xuru** [comment]
This totally makes sense.  Kind of over engineered it.  I'll make the changes, and see how it works.

**Comment by xuru** [comment]
@Suor Ok, ready for your review, and thanks for the pointers.

File: cacheops/query.py [hunk 1]
```diff
@@ -4,6 +4,8 @@
 import threading
 import six
 from random import random
+
+from django.db.models.manager import BaseManager
```
Comment (Suor) [inline cacheops/query.py]: Please place this right after `from django.db.models import Model` import.

File: tests/tests.py [hunk 2]
```diff
@@ -607,6 +606,12 @@ def test_312(self):
     def test_316(self):
         Category.objects.cache().annotate(num=Count('posts')).aggregate(total=Sum('num'))
 
+    def test_333(self):
+        # ensure that we can use a manager not derived from Manger
+        # (i.e. using manager.BaseManager.from_queryset)
+        mgr = CustomFromQSModel.objects
+        assert isinstance(mgr, CustomFromQSManager)
```
Comment (Suor) [inline tests/tests.py]: This doesn't really test anything. The old code will blow up in `install_cacheops()` call I suppose. You may just drop it.

File: cacheops/query.py (hunk 1 above)
Comment (xuru) [reply cacheops/query.py]: Done

**Pull request closed without merging**